src/Core/Managed/Shared/Extensibility/Implementation/External/ExceptionDetails_types.cs

```diff
@@ -77,5 +77,30 @@ protected ExceptionDetails(string fullName, string name)
             stack = "";
             parsedStack = new List<StackFrame>();
         }
+
+        /// <summary>
+        /// Creates a new instance of ExceptionDetails from a System.Exception and a parent ExceptionDetails.
+        /// </summary>
+        internal static ExceptionDetails CreateWithoutStackInfo(Exception exception, ExceptionDetails parentExceptionDetails)
```

Comment:
Moved the implementation to a third partial class to keep the pattern consistent.